### PR TITLE
add example for use with bot base

### LINKED ISSUE #7: Use deltachat-node-bot-base for nodejs-bot
We've been working partly in parallel regarding NodeJS. I implemented a small abstraction layer for NodeJS to use as bot library: https://github.com/deltachat-bot/deltachat-node-bot-base

I suggest to use that library for the NodeJS-bot in order to avoid repetition. [The README](https://github.com/deltachat-bot/deltachat-node-bot-base/blob/master/README.md) includes example code that explains how to use it.

@Simon-Laux If you agree but don't want to do it yourself, let me know and I'll take it into my hands.

### PR DESCRIPTION
**Pull request by Simon-Laux**
closes #7

### CODE REVIEW

**Comment by Simon-Laux** [comment]
Somehow it works only sometimes halfway. In its current state I can hardly recommend this to anybody. Simple is no good if it doesn't work in the end.

File: nodejs_bot_base/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { deltachat, log } = require('deltachat-node-bot-base')
+
+// Start the deltachat core engine and handle incoming messages.
+deltachat.start((chat, message) => {
+  log("Got a message: ", message.getText())
+  if (deltachat.getChatContacts(chat.getid()).size === 1) {
```
Comment (Simon-Laux) [inline nodejs_bot_base/index.js]:
```suggestion
  if (deltachat.getChatContacts(chat.getid()).length === 1) {
```
[].size doesn't exist in javascript.

**Comment by pabzm** [comment]
> Somehow it works only sometimes halfway. In its current state I can hardly recommend this to anybody. Simple is no good if it doesn't work in the end. So I'm considering to just close this.

Does this opinion still stand? Then you should close this.

If not, is the "wip" status still correct?

**Comment by Simon-Laux** [comment]
I fixed in the last commit, as far as I remember. 
I still don't love your attempt for simplification, but If you want to go with it I will help you with this example and I could probably do a commit to fix the types in the bot-base so that the user gets auto-completion.
So I vote for merging after you approve it.

**Comment by pabzm** [comment]
Thank you for being willing to help!

But to be honest, I don't think this example is necessary. It resembles the README of deltachat-node-bot-base, and its texts make clear that you don't really like the solution. As a reader it makes me think "why is this here, if they don't like it?"

I can approve the changes, if you are interested to merge it.

>  fix the types in the bot-base so that the user gets auto-completion

What types and which auto-completion do you mean? I don't understand the context.

File: nodejs_bot_base/README.MD [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Echo Bot - Nodejs Bot Base
+
+This is an makes use of `deltachat-node-bot-base` an abstraction Layer over [`deltachat-node`](https://github.com/deltachat/deltachat-node) which does the initial setup for you. It is opninionated (makes some design decissions for you, like the used config module) -> If you want more controll look at our other echo-bot example that uses `deltachat-node` directly: [Advanced Example](../nodejs)
```
Comment (Simon-Laux) [inline nodejs_bot_base/README.MD]:
```suggestion
This is an makes use of `deltachat-node-bot-base` an abstraction Layer over [`deltachat-node`](https://github.com/deltachat/deltachat-node) which does the initial setup for you. It is opinionated (makes some design decisions for you, like the used config module) -> If you want more control look at our other echo-bot example that uses `deltachat-node` directly: [Advanced Example](../nodejs)
```

**Comment by Simon-Laux** [comment]
> and its texts make clear that you don't really like the solution

nah... I feel that the text in the readme is fairly neutral and states facts, if you have a suggestion on how to improve the wording, please let me know. (there is even a typo in there so we need to correct that before merging anyway)

Also opinionated isn't a bad thing: https://prettier.io/ states it proudly on their website.

**Comment by Simon-Laux** [comment]
> What types and which auto-completion do you mean? I don't understand the context.

lets don't discuss this here, we can discuss this better in an av. The point is typescript typings so that your IDE can make good suggestions and also when I think about it we should add some-kind of documentation to that bot base. and think about reintroducing the constants at least as long as we don't have other ways to do checks that you do with them implemented.

**Pull request merged**